native~/Runtime/src/CesiumSimplePlanarEllipsoidCurveImpl.cpp

```diff
@@ -25,12 +25,12 @@ bool CesiumSimplePlanarEllipsoidCurveImpl::
           glm::dvec3(sourceEcef.x, sourceEcef.y, sourceEcef.z),
           glm::dvec3(destinationEcef.x, destinationEcef.y, destinationEcef.z));
   if (!flightPath.has_value()) {
-    this->_curve = nullptr;
+    this->_curve = std::nullopt;
     return false;
   }
 
-  this->_curve = std::make_unique<SimplePlanarEllipsoidCurve>(*flightPath);
-  return this->_curve != nullptr;
+  this->_curve = *flightPath;
+  return this->_curve.has_value();
```

Comment:
This function (and the one below) can be simplified quite a bit now that you're using `std::optional`. Something like this (not tested):

```cpp
  this->_curve =
      SimplePlanarEllipsoidCurve::fromEarthCenteredEarthFixedCoordinates(
          Ellipsoid::WGS84,
          glm::dvec3(sourceEcef.x, sourceEcef.y, sourceEcef.z),
          glm::dvec3(destinationEcef.x, destinationEcef.y, destinationEcef.z));
  return this->_curve.has_value();
```